csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -1397,6 +1406,30 @@ private module ReturnNodes {
     override string toStringImpl() { result = yrs.toString() }
   }
 
+  /**
+   * A synthesized `return` node for returned expressions inside `async` methods.
+   */
+  class AsyncReturnNode extends ReturnNode, NodeImpl, TAsyncReturnNode {
```

Comment:
At the moment we are not handling async streams. See https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/async-return-types#async-streams-with-iasyncenumerablet